Zw/feature/inputfield

zachwolek opened the pull request:
## Description of Changes
ShortenLinkPage is a parent component to CopyLink and InputField children components. The user has the ability to submit a link, the link submitted with userId to generate a shortened link, the request of POST /api/v1/users/:id/links?link={original link} returns JSON info in the turlink-be contract. User has the ability to copy the shortened link in the CopyLink component. 

## What are the relevant tickets (if any) 
Closes #11 for input field of original link with placeholder
Closes #12 for "Shorten Link" button creation
Closes #13 to display shortened link and save JSON info to state
Closes #15 to submit POST request and return JSON info

## Screenshots (if applicable)
N/A

## Testing
N/A

## Checklist
- [X ] The code follows the project's coding standards.
- [ ] Unit tests have been added or updated to cover the changes.
- [ ] Documentation has been updated to reflect the changes (if applicable).
- [ X] The code compiles without errors.
- [ ] The changes have been tested locally and pass all relevant tests.
- [ ] All new and existing tests pass.
- [ X ] The pull request has been reviewed by at least one other contributor.

## Reviewer Instructions
Branch has already been merged with turingschool/RG/feature/dashboard-layout and should not encounter much conflict

## Deployment Notes
N/A

## Additional Information
Component functionality is ready, component still needs
- Presentation of error messages from POST requests
- CSS cleanup from merge
- Mobile testing to address the final need on #25 
- Cypress testing

reesegreen2014 commented:
Thank you for your work on this @zachwolek! I have a few questions:

1. There is some commented out code on line 11 in the `ShortenLinkPage.tsx` file. If this is not being used, can you please remove it?
2. There are a few code compilation errors in the terminal when running the local host with the changes from this branch. Can you please run `npm start` on your end and observe the errors in the terminal and correct them where necessary? If the error is stemming from or related to a file outside of the scope of your folders, that is okay. Those might/can be resolved once we connect all the components together. 
3. Is there any current or planned error handling in place that displays appropriate error messages to the user should the application not behave as expected (error with POST (request or response), incorrect link format, etc)? Not a huge priority for the sake of completing the MVP, but could be something to implement during part 2 of the capstone.

zachwolek commented:
Thank you for your feedback @reesegreen2014!

1) I have deleted the commented out code
2) Let me know if you still encounter compilation errors. After viewing the console, I am to the current understanding that this was a result of the CopyLink.tsx input field not having an "onChange" function attached, as we do not want to give the user the ability to accidentally type over the shortened link we provide. 
I have changed the value associated with this input to `defaultValue={shortenedLink}` and have added the input field is `readOnly`, which should prevent the errors expecting an onChange from the input field.
3) Yes. To my understanding in the Zoom team meeting, the decision to merge as early as possible was prioritized as long as functionality was in place. This remaining need is already outlined in the first bullet point of "Additional Information", my apologies if my language was not clear specifying.
